### DISCUSSION TITLE
Hide Monitor from Computer When on Different Input?

### DISCUSSION THREAD

**Discussion by 8va** [MonitorControl/MonitorControl#994] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues

### Is your feature request related to a problem? Please describe

I often switch between devices/inputs on my external monitor, and when I do that, I often want to use my MacBook Pro just with its own screen. The MBP still thinks the external display is connected and thus windows and display settings need to be changed every time, and then changed back when I go back to using the external monitor as a second display for the MBP.

### Describe the solution you'd like

This might be beyond the scope of MonitorControl, but I wonder if it would be possible to create a setting that will hide the external monitor from the MacBook Pro when the external monitor is using a different input.

### Describe alternatives you've considered

The only alternative I can think of is changing System Preference's Display settings back and forth every time I use another display input.

Unplugging the display cable isn't a great solution either, as it's really connected through a Thunderbolt hub beneath my desk.

**Comment by waydabber** [comment]
Unfortunately there is no way to fully remove a display from the display configuration in macOS. In the past (earlier macOS builds) there was an unsupported API for this but it no longer works (fun fact: this as infamous for rendering some macs inoperable). The only thing we can do is to mirror the internal display to the unneeded external screen thus effectively "hiding" it behind the main screen. We could easily implement a function like this. Alternatively if you download my other app [BetterDummy](https://github.com/waydabber/BetterDummy), it has a simple menu option to configure mirroring quickly - you'd just have to click on this to hide your external screen behind your internal screen, effectively making it go away:

<img width="508" alt="Screen Shot 2022-04-07 at 19 53 59" src="https://user-images.githubusercontent.com/37590873/162266411-205f40e2-c304-421e-90de-61df0248f80c.png">

**Reply by 8va** [reply]
Awesome! I'd greatly appreciate that feature (having the computer switch to mirror mode automatically when the external display changes to a different input).

BetterDummy seems like a decent workaround, thanks for the suggestion. I'd still prefer it to be an automatic process (and not have to add another icon to my menubar), but I might give it a try until then.

**Reply by waydabber** [reply]
I'll add this feature to BetterDummy - https://github.com/waydabber/BetterDummy/issues/474
